include/ibamr/INSStaggeredHierarchyIntegrator.h

```diff
@@ -232,6 +240,21 @@ class INSStaggeredHierarchyIntegrator : public INSHierarchyIntegrator
     void removeNullSpace(const SAMRAI::tbox::Pointer<SAMRAI::solv::SAMRAIVectorReal<NDIM, double> >& sol_vec);
 
 protected:
+    /*!
+     * L1 norm of the fluid velocity before regridding.
+     */
+    double d_div_U_norm_1_pre = 0.0;
```

Comment:
Could/should these be moved into the base class?

Reply:
Technically yes, but since `INSCollocatedHierarchyIntegrator` doesn't use them I thought it would be slightly better to duplicate them across two inheriting classes.

I tried to do this and to also move the 'check to see if the divergence grew' code into the base class. Unfortunately, since inheriting classes use different cell variables (e.g., `FaceVariable` instead of `CellVariable`) its not possible to make some calls to `HierarchyMathOps` without complete type information, which is why I opened #526.

Reply:
I figured the duplication was slightly better than the alternatives.